sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -552,18 +570,87 @@ private void validateApplicationType() {
     }
 
     private boolean isValidApplicationType(AadApplicationType configured, AadApplicationType inferred) {
-        return inferred == configured || inferred == AadApplicationType.RESOURCE_SERVER_WITH_OBO;
+        return inferred == configured || inferred == RESOURCE_SERVER_WITH_OBO;
     }
 
     private void validateAuthorizationClientProperties(String registrationId,
                                                        AuthorizationClientProperties properties) {
+        String grantType = decideGrantType(registrationId, properties, applicationType);
+        if (grantType != null) {
+            // Validate authorization grant grantType
+            validateAuthorizationGrantType(registrationId, grantType, properties);
+        }
+
+        // Validate scopes.
+        List<String> scopes = validateScopes(registrationId, properties);
+
+        addNecessaryScopesForAuhtorizationCodeClients(properties, scopes);
+    }
+
+    /**
+     * Add necessary scopes for authorization_code clients.
+     *
+     * https://docs.microsoft.com/graph/permissions-reference#remarks-17
+     * https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
+     *
+     * @param properties AuthorizationClientProperties
+     * @param scopes scopes for authorization_code clients.
+     */
+    private void addNecessaryScopesForAuhtorizationCodeClients(AuthorizationClientProperties properties,
+                                                               List<String> scopes) {
+        if (properties.getAuthorizationGrantType().getValue().equals(AUTHORIZATION_CODE.getValue())) {
```

Comment:
Can we use `properties.getAuthorizationGrantType() == AUTHORIZATION_CODE)` here?

Reply:
ok